tools/indexer/pkg/ethereum/cleint.go

```diff
@@ -0,0 +1,191 @@
+package ethereum
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/primev/mev-commit/indexer/pkg/config"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	httputil "github.com/primev/mev-commit/indexer/pkg/http"
+)
+
+const optInABIJSON = `[
```

Comment:
I gave this comment in the last PR as well. We can use the generated client in `contracts-abi`. All it needs is ethereum client and it will handle all the data types.